src/main/kotlin/com/github/edeandrea/xjcplugin/type/Xjc.kt

```diff
@@ -129,51 +144,86 @@ open class Xjc : DefaultTask() {
-			val optionsMap = mutableMapOf(
-				"destdir" to this.schemaGenDir.absolutePath,
-				"extension" to true,
-				"schema" to schemaFile,
-				"language" to language
-			)
-
-			if (this.javaPackageName.isNotBlank()) {
-				optionsMap["package"] = this.javaPackageName
+				val language = if (schemaFile.name.endsWith("wsdl")) "WSDL" else "XMLSCHEMA"
+				processXjc(language, schemaFile, null)
 			}
+		}
+	}
 
-			if (this.bindingFile != null) {
-				optionsMap["binding"] = this.bindingFile
-			}
+	private fun processXjc(language: String, schemaFile: File?, schemaFiles: FileCollection?) {
+		val optionsMap = mutableMapOf(
+			"destdir" to this.schemaGenDir.absolutePath,
+			"extension" to true,
+			"language" to language
+		)
 
-			if (this.additionalXjcOptions.isNotEmpty()) {
-				this.additionalXjcOptions.forEach { (k, v) -> optionsMap.putIfAbsent(k, v)}
-			}
+		if (this.javaPackageName.isNotBlank()) {
+			optionsMap["package"] = this.javaPackageName
+		}
+
+		if (this.bindingFile != null) {
+			optionsMap["binding"] = this.bindingFile
+		}
+
+		if (this.additionalXjcOptions.isNotEmpty()) {
+			this.additionalXjcOptions.forEach { (k, v) -> optionsMap.putIfAbsent(k, v) }
+		}
 
-			val commandLineArgs = parseCommandLineArgs()
+		val commandLineArgs = parseCommandLineArgs()
 
-			LOGGER.lifecycle("Running XJC compiler for schema $schemaFile")
-			LOGGER.lifecycle("\tOptions:")
-			optionsMap.forEach { (k, v) ->
-				LOGGER.lifecycle("\t\t$k = $v")
+		LOGGER.lifecycle("Running XJC compiler for schema(s) ${schemaFile ?: schemaFiles?.asPath}")
+		LOGGER.lifecycle("\tOptions:")
+		optionsMap.forEach { (k, v) ->
+			LOGGER.lifecycle("\t\t$k = $v")
+		}
+
+		if (commandLineArgs.isNotEmpty()) {
+			LOGGER.lifecycle("\tCommand Line Args:")
+			commandLineArgs.forEach { arg ->
+				LOGGER.lifecycle("\t\t$arg")
 			}
+		}
+
+		if (schemaFile != null) {
+			optionsMap["schema"] = schemaFile
 
-			if (commandLineArgs.isNotEmpty()) {
-				LOGGER.lifecycle("\tCommand Line Args:")
-				commandLineArgs.forEach { arg ->
-					LOGGER.lifecycle("\t\t$arg")
+			ant.withGroovyBuilder {
+				"xjc"(optionsMap) {
+					commandLineArgs.forEach { arg ->
+						"arg"("value" to arg)
+					}
 				}
 			}
-
+		}
+		else {
 			ant.withGroovyBuilder {
 				"xjc"(optionsMap) {
 					commandLineArgs.forEach { arg ->
 						"arg"("value" to arg)
 					}
+					schemaFiles!!.forEach {
+						"schema"("file" to it)
+					}
 				}
 			}
 		}
 	}
 }
+
+fun guessLanguageAndDetectMixedSchemaTypes(schemaFiles: FileCollection): String {
+	return schemaFiles.fold("", ::detectMixedSchemaType)
+}
+
+private fun detectMixedSchemaType(language: String, file: File): String {
+	if (!"XMLSCHEMA".equals(language) && file.name.endsWith("wsdl")) return "WSDL"
+	else if (!"WSDL".equals(language) && file.name.endsWith("xsd")) return "XMLSCHEMA"
+	throw GradleException("With one-ass mode it is not possible to generate classes from XMLSCHEMA and WSDL schemas. Use either XMLSCHEMA or WSDL files.")
```

Comment:
Can you fix this typo (`one-ass` should be `one-pass`)?

Reply:
Fixed also